src/wallet/wallet2.cpp

```diff
@@ -12901,6 +12902,74 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw) const
+{
+  if (m_export_format == ExportFormat::Hex)
+  {
+    std::string h = epee::string_tools::buff_to_hex_nodelimer(raw);
+
+    // Split the long string into multiple lines, similarly to how openssl
+    // prints certificates.
+    for (size_t i = 63; i < h.length(); i += 64)
+    {
+      h.insert(i, 1, '\n');
+    }
```

Comment:
That's going to be a lot of memcpy. Not sure it's worth doing anything about given there's a disk write later though.

Reply:
TBH I can't think of a solution that doesn't require copying of every chunk. We could copy _only_ that chunk in every iteration if we didn't do it in place but that would be at the cost of less readable code. And as you said, the bottleneck is likely in the disk write anyway.

Reply:
Yes, let's leave it like that. I can think of a solution in O(N) but it makes things more complicated. We can always change it later if it turns out there is a significant slowdown.